core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -830,6 +830,7 @@ class LogManager(logDirs: Seq[File],
         cleaner.abortCleaning(topicPartition)
         cleaner.updateCheckpoints(removedLog.dir.getParentFile)
       }
+      removedLog.close()
```

Comment:
this helped clearing the message that the directory cannot be renamed. First we close it entirely, then we can rename the dir